src/hotspot/share/runtime/stubRoutines.hpp

```diff
@@ -452,11 +453,13 @@ class StubRoutines: AllStatic {
   static address hf2f_adr()            { return _hf2f; }
 
   static jshort f2hf(jfloat x) {
+    MACOS_AARCH64_ONLY(ThreadWXEnable wx(WXExec, Thread::current());) // About to call into code cache
     assert(_f2hf != nullptr, "stub is not implemented on this platform");
```

Comment:
Would it make sense to have the assert as first thing in the method?

Reply:
Yes, probably a better idea to make sure the address (of the stub) being accessed is valid before making that memory region executable. Will make changes and update. Thanks!

Reply:
Done

Reply:
Flipping to execute and back during compilation seems to me to be a bad idea. I'd do it in inline asm instead.

Reply:
Thank you! That looks to be outside the scope of this task but I will definitely look into it.